pkg/chains/storage/grafeas/grafeas.go

```diff
@@ -214,21 +213,42 @@ func (b *Backend) createNote(ctx context.Context, tr *v1beta1.TaskRun, opts conf
 		)
 	}
 
-	// for taskrun: BuildNote
-	return b.client.CreateNote(ctx,
-		&pb.CreateNoteRequest{
-			Parent: b.getProjectPath(),
-			NoteId: fmt.Sprintf(buildNoteNameFormat, noteID),
-			Note: &pb.Note{
-				ShortDescription: "Build Provenance Note",
-				Type: &pb.Note_Build{
-					Build: &pb.BuildNote{
-						BuilderVersion: tr.GetGroupVersionKind().GroupVersion().String(),
+	switch obj.GetObject().(type) {
```

Comment:
Seems the only fields that differ here are `NoteId` and `ShortDescription`, would we want to switch on just those to reduce duplicated code? This is very much a nitpick, i'm also fine with how it is now.

Reply:
Good point. Happy to extract this out. Thanks @bcaton85 for reviewing the PR.